go-controller/pkg/cni/cniserver.go

```diff
@@ -102,6 +110,22 @@ func NewCNIServer(rundir string, useOVSExternalIDs bool, factory factory.NodeWat
 	return s, nil
 }
 
+// cancelPodAddsForOldMACs cancels and pod add requests that do not match
+// the given MAC address.
+func (s *Server) cancelPodAddsForOldMACs(pod *kapi.Pod, podMAC string) {
+	s.runningSandboxAddsLock.Lock()
+	defer s.runningSandboxAddsLock.Unlock()
+
+	for _, req := range s.runningSandboxAdds {
+		// Cancel sandbox requests for this pod that have a MAC set
+		// which doesn't match the expected MAC
+		if req.PodNamespace == pod.Namespace && req.PodName == pod.Name && req.MAC != "" && req.MAC != podMAC {
```

Comment:
why do you need to compare req.MAC  != ""? Is it because of this:
```	if err := s.startSandboxRequest(req); err != nil {
		return nil, err
	}
	defer s.finishSandboxRequest(req)

	useOVSExternalIDs := false
	if atomic.LoadInt32(&s.useOVSExternalIDs) > 0 {
		useOVSExternalIDs = true
	}
	result, err := s.requestFunc(req, s.podLister, useOVSExternalIDs, s.kclient)
```

There is a period of time when the request is added to the map before the request has been processed so may have no mac?